packages/abi-coder/src/json-abi.ts

```diff
@@ -111,11 +132,19 @@ export class ABI {
   }
 
   static unflatten(jsonAbi: JsonAbi) {
-    if (Array.isArray(jsonAbi)) {
-      return jsonAbi as ReadonlyArray<JsonAbiFragment>;
+    if (isFlatJsonAbi(jsonAbi)) {
+      const abi = new ABI(jsonAbi);
+      return abi.unflatten();
     }
-    const abi = new ABI(jsonAbi as JsonFlatAbi);
-    return abi.unflatten();
+
+    return jsonAbi;
+  }
+
+  unflattenLoggedTypes(): ReadonlyArray<JsonAbiLogFragment> {
+    return this.loggedTypes.map((loggedType) => ({
+      ...loggedType,
+      abiFragmentType: [this.parseLoggedType(loggedType)],
+    }));
```

Comment:
This is correct.  My understanding is that we `unflatten` it to make it work with the code. We really need to refactor the how the coders and ABI interact because this is where the tech debt is

Reply:
The idea behind keeping the nonthe  flat abi, was to keep possible partial definition of functions using objects. But once logs and other things are going to require types. We should remove it.